cdap-app-fabric/src/test/java/co/cask/cdap/internal/profile/ProfileServiceTest.java

```diff
@@ -412,4 +423,60 @@ public void testProfileCreationTime() throws Exception {
     profileService.saveProfile(myProfile, profile);
     Assert.assertEquals(creationTime, profileService.getProfile(myProfile).getCreatedTsSeconds());
   }
+
+  @Test
+  public void testProfileMetricsDeletion() throws Exception {
+    ProfileId myProfile = NamespaceId.DEFAULT.profile("MyProfile");
+    Profile profile = new Profile("MyProfile", Profile.NATIVE.getLabel(), Profile.NATIVE.getDescription(),
+                                  Profile.NATIVE.getScope(), Profile.NATIVE.getProvisioner());
+    ProgramRunId runId = NamespaceId.DEFAULT.app("myApp").workflow("myProgram").run(RunIds.generate());
+
+    // create and disable the profile
+    profileService.saveProfile(myProfile, profile);
+    profileService.disableProfile(myProfile);
+
+    // emit some metrics
+    MetricsCollectionService metricService = injector.getInstance(MetricsCollectionService.class);
+    MetricsContext metricsContext = metricService.getContext(getMetricsTags(runId, myProfile));
+    metricsContext.increment(Constants.Metrics.Program.PROGRAM_NODE_MINUTES, 30L);
+
+    MetricStore metricStore = injector.getInstance(MetricStore.class);
+    Tasks.waitFor(30L, () -> getMetric(metricStore, runId, myProfile,
+                                      "system." + Constants.Metrics.Program.PROGRAM_NODE_MINUTES),
+                  10, TimeUnit.SECONDS);
+
+    // delete and verify the metrics are gone
+    profileService.deleteProfile(myProfile);
+    Tasks.waitFor(0L, () -> getMetric(metricStore, runId, myProfile,
+                                       "system." + Constants.Metrics.Program.PROGRAM_NODE_MINUTES),
+                  10, TimeUnit.SECONDS);
+  }
+
+  private Map<String, String> getMetricsTags(ProgramRunId programRunId, ProfileId profileId) {
+    return ImmutableMap.<String, String>builder()
+      .put(Constants.Metrics.Tag.PROFILE_SCOPE, profileId.getScope().name())
+      .put(Constants.Metrics.Tag.PROFILE, profileId.getProfile())
+      .put(Constants.Metrics.Tag.NAMESPACE, programRunId.getNamespace())
+      .put(Constants.Metrics.Tag.PROGRAM_TYPE, programRunId.getType().getPrettyName())
+      .put(Constants.Metrics.Tag.APP, programRunId.getApplication())
+      .put(Constants.Metrics.Tag.PROGRAM, programRunId.getProgram())
+      .put(Constants.Metrics.Tag.RUN_ID, programRunId.getRun())
+      .build();
+  }
+
+  private long getMetric(MetricStore metricStore, ProgramRunId programRunId, ProfileId profileId, String metricName) {
+    Map<String, String> tags = getMetricsTags(programRunId, profileId);
+
+    MetricDataQuery query = new MetricDataQuery(0, 0, Integer.MAX_VALUE, metricName, AggregationFunction.SUM,
```

Comment:
this pattern seems to be done in multiple places in cdap - can we reuse?

Reply:
I didn't mean the constructor, but the pattern where we get a single value from it.

Reply:
Yea, I noted in the JIRA that we should refactor that as well, we have multiple unit tests and also in some production code which are using the same pattern to fetch metrics